x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -151,6 +151,10 @@ private void doExecuteForked(Task task, EsqlQueryRequest request, ActionListener
 
     @Override
     public void execute(EsqlQueryRequest request, EsqlQueryTask task, ActionListener<EsqlQueryResponse> listener) {
+        // set EsqlExecutionInfo on async-search task so that it is accessible to GET _query/async while the query is still running
+        if (task instanceof EsqlQueryTask esqlQueryTask) {
```

Comment:
This check seems redundant - the function params already say `EsqlQueryTask task`

Reply:
Good catch. I used to have this in another place where I had to do the cast and when I moved it, I didn't notice that the cast wasn't needed. Thanks! Fixed in next push.